Sources/TSCBasic/Process.swift

```diff
@@ -23,6 +23,13 @@ import Dispatch
 
 import _Concurrency
 
+#if os(Windows)
+public typealias ProcessEnvironmentBlock = Dictionary<CaseInsensitiveString, String>
+#else
+public typealias ProcessEnvironmentBlock = Dictionary<String, String>
+#endif
```

Comment:
This will require #if os(Windows)` at the place of use. How about creating a new `ProcessEnvironment` type hiding these platform differences under the hood?

Reply:
I'm not sure I follow.  The typealias is meant to be the means for avoiding the `#if os` check at the use site.

Reply:
Ah, I see what you mean now.  Sure, that sounds like a good idea.

Reply:
Do you want me to keypath the type so that we can easily expose the other methods on the dictionary?

Reply:
Do you mean `@dynamicMemberLookup` with key paths or something else? IIUC `@dynamicMemberLookup` doesn't support proxying methods, only effectless properties. Maybe this new type could provide a read-only access to the underlying dictionary instead, while modifications are allowed through a custom subscript doing the normalization itself?

Reply:
Ah, bleh; okay.  In that case, what we have currently is reasonable.  We can extend this later if needed.